contracts/wrapperContracts/WrapConversionRate.sol

```diff
@@ -2,246 +2,89 @@ pragma solidity 0.4.18;
-        return(signatures);
-    }
-
-    function performAddToken() internal {
-        conversionRates.addToken(addTokenData.token);
+        conversionRates.addToken(token);
 
         conversionRates.addOperator(this);
 
         //token control info
         conversionRates.setTokenControlInfo(
-            addTokenData.token,
-            addTokenData.minimalResolution,
-            addTokenData.maxPerBlockImbalance,
-            addTokenData.maxTotalImbalance
+            token,
+            minRecordResolution,
+            maxPerBlockImbalance,
+            maxTotalImbalance
         );
 
         //step functions
         int[] memory zeroArr = new int[](1);
         zeroArr[0] = 0;
 
-        conversionRates.setQtyStepFunction(addTokenData.token, zeroArr, zeroArr, zeroArr, zeroArr);
-        conversionRates.setImbalanceStepFunction(addTokenData.token, zeroArr, zeroArr, zeroArr, zeroArr);
+        conversionRates.setQtyStepFunction(token, zeroArr, zeroArr, zeroArr, zeroArr);
+        conversionRates.setImbalanceStepFunction(token, zeroArr, zeroArr, zeroArr, zeroArr);
 
-        conversionRates.enableTokenTrade(addTokenData.token);
+        conversionRates.enableTokenTrade(token);
 
         conversionRates.removeOperator(this);
     }
 
-    function performSetTokenControlInfo() internal {
-        require(tokenControlInfoData.tokens.length == tokenControlInfoData.perBlockImbalance.length);
-        require(tokenControlInfoData.tokens.length == tokenControlInfoData.maxTotalImbalance.length);
+    //set token control info
+    ////////////////////////
+    function setTokenControlData(ERC20[] tokens, uint[] maxPerBlockImbalanceValues, uint[] maxTotalImbalanceValues)
+        public
+        onlyAdmin
+    {
+        require(maxPerBlockImbalanceValues.length == tokens.length);
+        require(maxTotalImbalanceValues.length == tokens.length);
 
         uint minRecordResolution;
 
-        for (uint i = 0; i < tokenControlInfoData.tokens.length; i++) {
+        for (uint i = 0; i < tokens.length; i++) {
             uint maxPerBlock;
             uint maxTotal;
             (minRecordResolution, maxPerBlock, maxTotal) =
-                conversionRates.getTokenControlInfo(tokenControlInfoData.tokens[i]);
+            conversionRates.getTokenControlInfo(tokens[i]);
             require(minRecordResolution != 0);
 
-            conversionRates.setTokenControlInfo(tokenControlInfoData.tokens[i],
+            conversionRates.setTokenControlInfo(tokens[i],
                 minRecordResolution,
-                tokenControlInfoData.perBlockImbalance[i],
-                tokenControlInfoData.maxTotalImbalance[i]);
+                 maxPerBlockImbalanceValues[i],
+                maxTotalImbalanceValues[i]);
         }
     }
-}
 
+    //valid duration blocks
+    ///////////////////////
+    function setValidDurationData(uint validDurationBlocks) public onlyAdmin {
+        require(validDurationBlocks > 5);
```

Comment:
Would it be better to set the `5` as a public constant? Or variable if we consider ETH2.0 :p

Reply:
possible.
will set public constant.

its all very internal stuff.